server.ts

```diff
@@ -374,9 +374,19 @@ function cacheCheck(req, res, next) {
   }
 
   // If cached copy exists, return it to the user.
-  if (cachedCopy) {
+  if (cachedCopy && cachedCopy.page) {
+    if (cachedCopy.headers && Array.isArray(environment.cache.serverSide.headers) && environment.cache.serverSide.headers.length > 0) {
+      environment.cache.serverSide.headers.forEach((header) => {
```

Comment:
@atarix83 : I'd rather we do this check during the caching process (currently it's being done when reading from the cache). That way we don't unnecessarily cache headers which we won't ever use.

So, for example, if the configuration says we are only caching `Link`, then we should only cache that single header...instead of caching all headers, but only using the `Link` header.

Reply:
done